client.socket.remoteAddress undefined!

### system info
Haraka | Haraka.js — Version: 2.8.12
Node | v6.9.4
OS | Linux mailgateway 4.4.0-59-generic #80-Ubuntu SMP Fri Jan 6 17:47:47 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
openssl | OpenSSL 1.0.2g  1 Mar 2016

### Observed behavior
using smtp_forward plugin to an Exchange server.
everytime on end of smtp_client connection the following logfile entry is created:

[ERROR] [-] [core] client.socket.remoteAddress undefined!

I think everything works fine, is it really an error?
or would logwarn() or lognotice() be more useful in smtp_client.js:151

--- a/smtp_client.js
+++ b/smtp_client.js
@@ -72,11 +72,11 @@ function SMTPClient (port, host, connect_timeout, idle_timeout) {
                 client.emit('auth_username');
                 return;
             }
-            else if (code.match(/^3/) && msg === 'UGFzc3dvcmQ6') {
+            if (code.match(/^3/) && msg === 'UGFzc3dvcmQ6') {
                 client.emit('auth_password');
                 return;
-            } else if (code.match(/^2/) && client.authenticating) {
-                //TODO: logging
+            }
+            if (code.match(/^2/) && client.authenticating) {
                 logger.loginfo('AUTHENTICATED');
                 client.authenticating = false;
                 client.authenticated = true;
@@ -143,13 +143,12 @@ function SMTPClient (port, host, connect_timeout, idle_timeout) {
     client.socket.on('connect', function () {
         // Remove connection timeout and set idle timeout
         client.socket.setTimeout(((idle_timeout) ? idle_timeout : 300) * 1000);
-        if (client.socket.remoteAddress) {
+        if (!client.socket.remoteAddress) {
             // "Value may be undefined if the socket is destroyed"
-            client.remote_ip = ipaddr.process(client.socket.remoteAddress).toString();
+            logger.logdebug('socket.remoteAddress undefined');
+            return;
         }
-        else {
-            logger.logerror('client.socket.remoteAddress undefined!');
-        }
+        client.remote_ip = ipaddr.process(client.socket.remoteAddress).toString();
     });
 
     var closed = function (msg) {

